src/Mooc/Courses/Application/FindAll/AllCoursesResponse.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+
+namespace CodelyTv\Mooc\Courses\Application\FindAll;
+
+
+use CodelyTv\Mooc\Courses\Domain\Courses;
+use CodelyTv\Shared\Domain\Bus\Query\Response;
+
+final class AllCoursesResponse implements Response
+{
+    private array $content;
+
+    public function __construct(Courses $courses)
+    {
+        $iterator = $courses->getIterator();
+        while ($iterator->valid()) {
+            $course = $iterator->current();
+            $this->content[] = [
+                'id' => $course->id(),
+                'name' => $course->name(),
+            ];
+            $iterator->next();
+        }
```

Comment:
I would move this logic to a `ResponseTransformer` or to the application service